crypto/src/main/java/net/corda/v5/crypto/KeyUtils.java

```diff
@@ -42,17 +42,16 @@ private KeyUtils() {
      * @param otherKeys An {@link Iterable} sequence of {@link PublicKey}.
      * @return True if <code>key</code> is in otherKeys.
      */
-    public static boolean isKeyInSet(@NotNull PublicKey key, @NotNull Iterable<PublicKey> otherKeys) {
+    public static boolean isKeyInSet(@NotNull PublicKey key, @NotNull Set<PublicKey> otherKeys) {
         if (key instanceof CompositeKey) {
             CompositeKey compositeKey = (CompositeKey) key;
             Set<PublicKey> leafKeys = compositeKey.getLeafKeys();
+            // TODO Here we could be checking which is set has less items and iterate through that
             for (PublicKey otherKey : otherKeys) {
```

Comment:
Should perhaps use https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Set.html#retainAll(java.util.Collection) to do set intersection and if non-empty return true

Reply:
Done